modules/xmpp/JingleSessionPC.js

```diff
@@ -19,1476 +19,1561 @@ import * as JingleSessionState from "./JingleSessionState";
+                    lines += '\r\n';
+                });
+            });
+            currentRemoteSdp.media.forEach(function(media, idx) {
+                if (!SDPUtil.find_line(media, 'a=mid:' + name))
+                    return;
+                if (!addSsrcInfo[idx]) {
+                    addSsrcInfo[idx] = '';
+                }
+                addSsrcInfo[idx] += lines;
             });
         });
-        currentRemoteSdp.media.forEach(function(media, idx) {
-            if (!SDPUtil.find_line(media, 'a=mid:' + name))
-                return;
-            if (!addSsrcInfo[idx]) {
-                addSsrcInfo[idx] = '';
-            }
-            addSsrcInfo[idx] += lines;
-        });
-    });
-    return addSsrcInfo;
-};
-
-/**
- * Handles a Jingle source-add message for this Jingle session.
- * @param elem An array of Jingle "content" elements.
- */
-JingleSessionPC.prototype.addRemoteStream = function (elem) {
-    // FIXME: dirty waiting
-    if (!this.peerconnection.localDescription) {
-        logger.warn("addSource - localDescription not ready yet");
-        setTimeout(() => this.addRemoteStream(elem), 200);
-        return;
+        return addSsrcInfo;
     }
-    logger.log('Processing add remote stream');
-    logger.log('ICE connection state: ', this.peerconnection.iceConnectionState);
-
-    this.readSsrcInfo(elem);
-
-    let workFunction = (finishedCallback) => {
-        var sdp = new SDP(this.peerconnection.remoteDescription.sdp);
-        var mySdp = new SDP(this.peerconnection.localDescription.sdp);
-        let addSsrcInfo = this._parseSsrcInfoFromSourceAdd(elem, sdp);
-
-        let newRemoteSdp = this._processRemoteAddSource(addSsrcInfo);
-        this._renegotiate(newRemoteSdp)
-            .then(() => {
-                logger.info("Remote source-add processed");
-                var newSdp = new SDP(this.peerconnection.localDescription.sdp);
-                logger.log("SDPs", mySdp, newSdp);
-                this.notifyMySSRCUpdate(mySdp, newSdp);
-                finishedCallback();
-            }, (error) => {
-                logger.error("Error renegotiating after processing remote source-add: " + error);
-                finishedCallback(error);
-            });
-    };
-    this.modificationQueue.push(workFunction);
-};
 
-/**
- * Handles a Jingle source-remove message for this Jingle session.
- * @param elem An array of Jingle "content" elements.
- */
-JingleSessionPC.prototype.removeRemoteStream = function (elem) {
-    // FIXME: dirty waiting
-    if (!this.peerconnection.localDescription) {
-        logger.warn("removeSource - localDescription not ready yet");
-        setTimeout(() => this.removeRemoteStream(elem), 200);
-        return;
+    /**
+     * Handles a Jingle source-add message for this Jingle session.
+     * @param elem An array of Jingle "content" elements.
+     */
+    addRemoteStream (elem) {
+        // FIXME: dirty waiting
```

Comment:
fixme how? if there is no realistic way to fix it I vote we do away with the comment.